test/server/configuration_impl_test.cc

```diff
@@ -119,5 +120,38 @@ TEST(ConfigurationImplTest, BadListenerConfig) {
   EXPECT_THROW(config.initialize(*loader), Json::Exception);
 }
 
+TEST(ConfigurationImplTest, NotSetServiceClusterWhenLSTracing) {
+  std::string json = R"EOF(
+  {
+    "listeners" : [
+      {
+        "port" : 1234,
+        "filters": []
+      }
+    ],
+    "cluster_manager": {
+      "clusters": []
+    },
+    "tracing": {
+      "http": {
+        "driver": {
+          "type": "lightstep",
+          "access_token_file": "/etc/envoy/envoy.cfg"
+        }
+      }
+    }
+  }
+  )EOF";
+
+  Json::ObjectPtr loader = Json::Factory::LoadFromString(json);
+
+  NiceMock<Server::MockInstance> server;
+  std::string cluster_not_set = "";
+  EXPECT_CALL(server.local_info_, clusterName()).WillOnce(ReturnRef(cluster_not_set));
```

Comment:
Just modify server.local_info_.cluster_name_ directly. You don't need a new variable or the EXPECT_CALL.